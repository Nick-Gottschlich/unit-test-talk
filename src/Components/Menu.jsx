// menu.js
import React, { Component } from 'react';

export const MenuItem = ({ children }) => <li>{children}</li>;

export default class Menu extends Component {
  static Item = MenuItem;

  render() {
    return (
      <ul>{this.props.children}</ul>
    );
  }
}