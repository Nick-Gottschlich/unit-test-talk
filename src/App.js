import React from "react";

import Button from "./Components/Button";
import Menu from "./Components/Menu";
import TextBox from "./Components/TextBox";
import ApiExample from "./Components/ApiExample";

import "./App.css";

const App = () => {
  const textBoxTitle = "Text Box Title";
  const textBoxSubTitle = "Text Box Subtitle";
  const textBoxItems = [
    {
      key: "i1",
      displayName: "item1"
    },
    {
      key: "i2",
      displayName: "item2"
    },
    {
      key: "i3",
      displayName: "item3"
    }
  ]
  const textBoxClickHandler = () => {
    console.log("you done went and clicked on the text box")
  };

  return (
    <div className="App">
      <header className="App-header">
        <div>
          <Button toggled="toggled" unToggled="unToggled" />
          <Button toggled="toggled" unToggled="unToggled" />
        </div>
        <Menu>
          <Menu.Item>
            Item 1
          </Menu.Item>
          <Menu.Item>
            Item 2
          </Menu.Item>
        </Menu>
        <div style={{ "marginBottom": "10px" }}>
          <TextBox
            title={textBoxTitle}
            subtitle={textBoxSubTitle}
            options={textBoxItems}
            handleClick={textBoxClickHandler}
          />
        </div>
        <ApiExample />
      </header>
    </div>
  );
}

export default App;
