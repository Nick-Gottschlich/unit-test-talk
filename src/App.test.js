/* eslint-disable no-unused-expressions */
import React from "react";
import { shallow, mount } from "enzyme";
import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow'

import App from "./App";
import Menu from "./Components/Menu";

describe("unit tests for the app container", () => {
  xit("unit test lesson 1: true should be truthy... right?", () => {
    expect(true).toBeTruthy;
  });

  let wrapper;

  xit("unit test lesson 2: don't be so shallow (or do)", () => {
    wrapper = shallow(<App />);

    //debug() provides an easy way for us to "see" what enzyme sees
    console.log('SHALLOW:', wrapper.debug());

    wrapper = mount(<App />);

    console.log('MOUNT: ', wrapper.debug());
  });

  xit("unit test lesson 3: find those buttons!", () => {
    wrapper = shallow(<App />);

    console.log(wrapper.debug());

    const button = wrapper.find("Button");

    // console.log(button.debug());

    expect(button).toHaveLength(2);
  });

  xit("unit test lesson 4: but wait, what about the dot?", () => {
    wrapper = shallow(<App />);

    // console.log(wrapper.debug());

    const menu = wrapper.find("Menu");
    expect(menu).toHaveLength(1);

    // const menuItem = wrapper.find("Menu.Item");
    // expect(menuItem).toHaveLength(2);
    // This didn't work! Why not?
    // Because it actually looked for: <Menu className="Item" />

    // const menuItem = wrapper.find(<Menu.Item />);
    // expect(menuItem).toHaveLength(2);
    // So close!!

    // const menuItem = wrapper.find(Menu.Item);
    // expect(menuItem).toHaveLength(2);
    // Finally!
  })

  xit("unit tesson lesson 11: snapshots", () => {
    const tree = renderer
      .create(<App/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })

  xit("unit tesson lesson 12: shallow snapshots", () => {
    const renderer = new ShallowRenderer()
    const result = renderer.render(<App />)
    expect(result).toMatchSnapshot()
  })
});
