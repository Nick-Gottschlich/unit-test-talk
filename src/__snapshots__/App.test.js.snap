// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unit tests for the app container unit tesson lesson 11: snapshots 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <div>
      <button
        onClick={[Function]}
        style={
          Object {
            "height": "50px",
            "width": "200px",
          }
        }
      >
        unToggled
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "height": "50px",
            "width": "200px",
          }
        }
      >
        unToggled
      </button>
    </div>
    <ul>
      <li>
        Item 1
      </li>
      <li>
        Item 2
      </li>
    </ul>
    <div
      style={
        Object {
          "marginBottom": "10px",
        }
      }
    >
      <div
        onClick={[Function]}
        style={
          Object {
            "border": "1px solid purple",
            "color": "pink",
          }
        }
      >
        <h1>
          Text Box Title
        </h1>
        <h2>
          Text Box Subtitle
        </h2>
        <ul>
          <li>
            item1
          </li>
          <li>
            item2
          </li>
          <li>
            item3
          </li>
        </ul>
      </div>
    </div>
    <div
      style={
        Object {
          "border": "1px solid gold",
          "color": "cyan",
        }
      }
    >
      <div>
        <button
          className="apiObjectButton"
          onClick={[Function]}
        >
          Api Example! Click me!
        </button>
      </div>
      <div>
        <span
          className="apiObjectTitle"
        />
      </div>
    </div>
  </header>
</div>
`;

exports[`unit tests for the app container unit tesson lesson 12: shallow snapshots 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <div>
      <Button
        toggled="toggled"
        unToggled="unToggled"
      />
      <Button
        toggled="toggled"
        unToggled="unToggled"
      />
    </div>
    <Menu>
      <MenuItem>
        Item 1
      </MenuItem>
      <MenuItem>
        Item 2
      </MenuItem>
    </Menu>
    <div
      style={
        Object {
          "marginBottom": "10px",
        }
      }
    >
      <TextBox
        handleClick={[Function]}
        options={
          Array [
            Object {
              "displayName": "item1",
              "key": "i1",
            },
            Object {
              "displayName": "item2",
              "key": "i2",
            },
            Object {
              "displayName": "item3",
              "key": "i3",
            },
          ]
        }
        subtitle="Text Box Subtitle"
        title="Text Box Title"
      />
    </div>
    <TextBox />
  </header>
</div>
`;
